test/functional/legacy/argument_count_spec.lua

```diff
@@ -0,0 +1,47 @@
+-- Tests for :[count]argument! and :[count]argdelete
+
+local helpers = require('test.functional.helpers')
+local clear, execute, eq, eval =
+  helpers.clear, helpers.execute, helpers.eq, helpers.eval
+
+describe('argument_count', function()
+  setup(clear)
+
+  it('is working', function()
+    execute('%argd')
+    execute('argadd a b c d')
+    eq({'a', 'b', 'c', 'd'}, eval('argv()'))
+    execute('set hidden')
+    execute('let buffers = []')
+    execute('augroup TEST')
+    execute([[au BufEnter * call add(buffers, expand('%:t'))]])
+    execute('augroup END')
+    execute('$argu')
+    execute('$-argu')
+    execute('-argu')
+    execute('1argu')
+    execute('+2argu')
+    execute('augroup TEST')
+    execute('au!')
+    execute('augroup END')
+    eq({'d', 'c', 'b', 'a', 'c'}, eval('buffers'))
```

Comment:
A linebreak here (and at the other assertion points) would help with readability.